source/decorated.d

```diff
@@ -50,10 +50,18 @@ mixin template decorated(string name, alias fun)
             }
         }
 
-        static if (isCallable!(typeof(impl!(__traits(getAttributes, _internal_helper_))())))
-            return impl!(__traits(getAttributes, _internal_helper_))()();
-        else
-            return impl!(__traits(getAttributes, _internal_helper_))();
+        auto call = impl!(__traits(getAttributes, _internal_helper_))();
+        static if (isCallable!(typeof(call)))
+        {
+            import std.traits : ParameterDefaults;
+            static if (args.length)
+            {
+                import core.lifetime : forward;
+                return call(forward!args, ParameterDefaults!fun[args.length .. $]);
+            }
+            else return call(ParameterDefaults!fun);
+        }
+        else return call;
     }
 
     mixin("alias "~name~" = _internal_helper_;");
```

Comment:
Not really related to this change, but we should check conformance with identifier grammar token (something like `([_a-zA-Z][_a-zA-Z0-9]*`, possibly allowing extra spaces too) to avoid weird errors. What if some user adds a `;` by mistake? We can give better error messages and later integrate with https://github.com/dlang/dmd/pull/13511 .

Reply:
see line 5

Reply:
That is weak checking and can be seen as insecure, if someone introduces text with `-J`. e.g.: `justkidding; void foo() {} int fake` this can be fed to trick the compiler.

Reply:
done